### DISCUSSION TITLE
tmp_path fixture with session scope

### DISCUSSION THREAD

**Discussion by pawamoy** [pytest-dev/pytest#8603] [Q&A] [answered]
Is there an easier way to get a session-scoped `tmp_path` than this?

```python
@pytest.fixture(scope="session")
def tmp_path(tmp_path_factory):
    basetemp = tmp_path_factory.getbasetemp()
    basetemp = basetemp.relative_to(basetemp.parents[0])
    return tmp_path_factory.mktemp(basetemp)
```

I'm not even sure it will use the same directory structure as the built-in `tmp_path` 😕 

EDIT: I'm now using this instead:

```python
@pytest.fixture(scope="session")
def tmp_path(tmp_path_factory):
    return tmp_path_factory.mktemp("data")
```

**Comment by bluetech** [comment, marked as answer]
> I'm now using this instead:

Yes, that's the way. I recommend using a different name than `tmp_path` however, since that is already used by the builtin function-scoped fixture.

**Reply by pawamoy** [reply]
Alright, thank you @bluetech !
